services/src/main/java/org/keycloak/protocol/oidc/endpoints/LogoutEndpoint.java

```diff
@@ -371,7 +371,7 @@ public Response logoutConfirmAction() {
             logger.debugf("Failed verification during logout. logoutSessionId=%s, clientId=%s, tabId=%s",
                     logoutSession != null ? logoutSession.getParentSession().getId() : "unknown", clientId, tabId);
 
-            if (logoutSession == null || logoutSession.getClient().equals(SystemClientUtil.getSystemClient(logoutSession.getRealm()))) {
+            if (logoutSession != null && logoutSession.getClient().equals(SystemClientUtil.getSystemClient(logoutSession.getRealm()))) {
```

Comment:
One concern here: In case that logoutSession is expired (so `logoutSession == null`), but logout was triggered with "system" client, than the more correct behaviour might be still to skip showing the link though?

Could we fallback to `clientId` parameter in that case? I guess something like this (not 100% sure code is correct, just for the illustration):
```
String usedClientId = "account";
if (logoutSession != null) {
  usedClientId = logoutSession.getClient().getClientId();
} else if (clientId != null) {
  // fallback to parameter clientId
  usedClientId = clientId;
}
if (usedClientId != null && usedClientId.equals(SystemClientUtil.getSystemClient(logoutSession.getRealm()).getClientId()) {
  // skip link ...
}
```

Will probably require some small adjustements to the test (or maybe new test)?

Reply:
Make sense, let me try this.